src/testing/database.ts

```diff
@@ -0,0 +1,143 @@
+import Statement from "../database/statement";
+import Callable from "../database/callable";
+import { getInstance } from "../base";
+import Table from "../database/table";
+import View from "../database/view";
+
+const systemLibrary = `sample`;
+const sqlSchema = `"TestDelimiters"`;
+
+export const DatabaseSuite: TestSuite = {
+  name: `Database object query tests`,
+  tests: [
+    {name: `Create environment`, test: async () => {
+      try {
+        await JobManager.runSQL(`call qsys.create_sql_sample('SAMPLE')`);
+      } catch (e) {
+        console.log(`Probably exists.`);
+      } finally {};
+
+      try {
+        const testDelimStatements = [
+          `DROP SCHEMA "TestDelimiters";`,
+          ``,
+          `CREATE SCHEMA "TestDelimiters";`,
+          `  `,
+          `CREATE TYPE "TestDelimiters"."Delimited Type" AS BIGINT;`,
+          `  `,
+          `CREATE TABLE "TestDelimiters"."Delimited Table" ("Delimited Column" INTEGER DEFAULT NULL, `,
+          `CONSTRAINT "TestDelimiters"."Delimited Key" PRIMARY KEY ("Delimited Column"));`,
+          ``,
+          `CREATE INDEX "TestDelimiters"."Delimited Index" ON "TestDelimiters"."Delimited Table" ("Delimited Column" ASC);`,
+          ``,
+          `CREATE TRIGGER "TestDelimiters"."Delimited Trigger" AFTER INSERT ON "TestDelimiters"."Delimited Table" `,
+          `FOR EACH ROW MODE DB2SQL BEGIN DECLARE X INT; END;`,
+          ``,
+          `CREATE VIEW "TestDelimiters"."Delimited View" ("Delimited Column") AS SELECT * FROM "TestDelimiters"."Delimited Table";`,
+          ``,
+          `CREATE ALIAS "TestDelimiters"."Delimited Alias" FOR "TestDelimiters"."Delimited Table";`,
+          ``,
+          `CREATE FUNCTION "TestDelimiters"."Delimited Function" ("Delimited Parameter" INTEGER) `,
+          `RETURNS INTEGER LANGUAGE SQL BEGIN RETURN "Delimited Parameter"; END;`,
+          ``,
+          `CREATE PROCEDURE "TestDelimiters"."Delimited Procedure" (INOUT "Delimited Parameter" INTEGER) `,
+          `LANGUAGE SQL BEGIN SET "Delimited Parameter" = 13; END;`,
+          ``,
+          `CREATE VARIABLE "TestDelimiters"."Delimited Global Variable" BIGINT;`,
+        ]
+        
+        // Server component doesn't support running multiple queries in one request
+        await getInstance().getContent().runSQL(testDelimStatements.join(`\n`));
+      } catch (e) {
+        console.log(`Possible fail`);
+      
+      } finally {};
+    }},
+
+    {name: `Delim name tests`, test: async () => {
+      const systemDelim = Statement.delimName(systemLibrary);
+      assert.strictEqual(systemDelim, `SAMPLE`);
+
+      const sqlDelim = Statement.delimName(sqlSchema);
+      assert.strictEqual(sqlDelim, sqlSchema); // No change expect
+
+      const withSpace = Statement.delimName(`my object`);
+      assert.strictEqual(withSpace, `"my object"`);
+
+      const longName = Statement.delimName(`create_sql_sample`);
+      assert.strictEqual(longName, `"create_sql_sample"`);
+    }},
```

Comment:
See my comment in `database/statement.ts`
We should add comments here to make it clear what we are testing.
This test is expecting the return value be delimited, indicating that the input value would have come from an SQL catalog.

Reply:
Test updated.